Modules/Database/Database.lua

```diff
@@ -18,49 +41,69 @@ function DB:Initialize()
     if type(CLM_DB[self.server_faction_guild]) ~= "table" then
         CLM_DB[self.server_faction_guild] = {}
     end
-    if type(CLM_DB[self.server_faction_guild]['personal']) ~= "table" then
-        CLM_DB[self.server_faction_guild]['personal'] = {}
+    if type(CLM_DB[self.server_faction_guild][DB_NAME_PERSONAL]) ~= "table" then
+        CLM_DB[self.server_faction_guild][DB_NAME_PERSONAL] = {}
     end
-    if type(CLM_DB[self.server_faction_guild]['personal']['gui']) ~= "table" then
-        CLM_DB[self.server_faction_guild]['personal']['gui'] = {}
+    if type(CLM_DB[self.server_faction_guild][DB_NAME_PERSONAL][DB_NAME_GUI]) ~= "table" then
+        CLM_DB[self.server_faction_guild][DB_NAME_PERSONAL][DB_NAME_GUI] = {}
     end
-    if type(CLM_DB[self.server_faction_guild]['guild']) ~= "table" then
-        CLM_DB[self.server_faction_guild]['guild'] = {}
+    if type(CLM_DB[self.server_faction_guild][DB_NAME_GUILD]) ~= "table" then
+        CLM_DB[self.server_faction_guild][DB_NAME_GUILD] = {}
     end
-    if type(CLM_DB[self.server_faction_guild]['raid']) ~= "table" then
-        CLM_DB[self.server_faction_guild]['raid'] = {}
+    if type(CLM_DB[self.server_faction_guild][DB_NAME_RAID]) ~= "table" then
+        CLM_DB[self.server_faction_guild][DB_NAME_RAID] = {}
     end
-    if type(CLM_DB[self.server_faction_guild]['ledger']) ~= "table" then
-        CLM_DB[self.server_faction_guild]['ledger'] = {}
+    if type(CLM_DB[self.server_faction_guild][DB_NAME_LEDGER]) ~= "table" then
+        CLM_DB[self.server_faction_guild][DB_NAME_LEDGER] = {}
     end
 end
 
 function DB:Global()
-    return CLM_DB['global']
+    return CLM_DB[DB_NAME_GLOBAL]
 end
 
 function DB:Logger()
-    return CLM_DB['global']['logger']
+    return CLM_DB[DB_NAME_GLOBAL][DB_NAME_LOGGER]
 end
 
 function DB:Server()
     return CLM_DB[self.server_faction_guild]
 end
 
-function DB:Personal()
-    return CLM_DB[self.server_faction_guild]['personal']
+function DB:Personal(table, schema)
```

Comment:
This should be split into 2 separate functions. Without a table argument it does something totally different than with.

Reply:
As discussed on discord removed this return and added an assertion